### PR TITLE
Close socket on error in NetConnect

### PR DESCRIPTION
**Pull request by embhorn**
Socket opened during NetConnect should be closed if there was an error during `SOCK_CONNECT` (except `MQTT_CODE_CONTINUE`)

### CODE REVIEW

File: examples/mqttnet.c [hunk 1]
```diff
@@ -540,7 +541,8 @@ static int NetConnect(void *context, const char* host, word16 port,
 
 exit:
     /* Show error */
-    if (rc != 0) {
+    if ((rc != 0) && (rc != MQTT_CODE_CONTINUE)) {
+        SOCK_CLOSE(sock->fd);
```
Comment (efieleke-tausight) [inline examples/mqttnet.c:545]: Is it safe to call SOCK_CLOSE if it was never opened successfully? That will happen if the code reaches line 448, 471 or 479.

File: examples/mqttnet.c [hunk 2]
```diff
@@ -630,7 +632,7 @@ static int SN_NetConnect(void *context, const char* host, word16 port,
 
   exit:
     /* Show error */
-    if (rc != 0) {
+    if ((rc != 0) && (rc != MQTT_CODE_CONTINUE)) {
```
Comment (efieleke-tausight) [inline examples/mqttnet.c:635]: Similar question here.

**Comment by embhorn** [comment]
@efieleke-tausight 

Interesting, yes, I changed it to use the existing disconnect which checks for a valid socket descriptor.

**Review by efieleke-tausight — APPROVED** [review]
Looks good to me!

File: examples/mqttnet.c [hunk 3]
```diff
@@ -410,6 +411,20 @@ static void tcp_set_nonblocking(SOCKET_T* sockfd)
 #endif /* WOLFMQTT_NONBLOCK */
 #endif /* !WOLFMQTT_NO_TIMEOUT */
 
+static int NetDisconnect(void *context)
+{
+    SocketContext *sock = (SocketContext*)context;
+    if (sock) {
+        if (sock->fd != SOCKET_INVALID) {
+            SOCK_CLOSE(sock->fd);
+            sock->fd = -1;
```
Comment (dgarske) [inline examples/mqttnet.c:420]: Let's fix this so `sock->fd = SOCKET_INVALID`.

**Review by dgarske — CHANGES_REQUESTED** [review]

**Review by dgarske — APPROVED** [review]

**Pull request merged**